src/ledger/LedgerTxn.cpp

```diff
@@ -1196,18 +1196,43 @@ LedgerTxnRoot::commitChild(EntryIterator iter)
 struct BulkLedgerEntryChangeAccumulator
 {
 
+    std::vector<LedgerEntry> AccountsToUpsert;
+    std::vector<LedgerKey> AccountsToDelete;
+
     // Accumulate any sufficiently-simple cases into our buffers, returning
     // true if the entry was accumulated, false otherwise (in which case the
     // entry will be committed on its own, normally).
     bool
     accumulate(EntryIterator const& iter)
     {
+        if (iter.key().type() == ACCOUNT)
+        {
+            if (iter.entryExists())
+            {
+                // Accounts with additional signers are not very common at
+                // present (less than 0.1% of accounts) and it is more
+                // complex to update signers in bulk, so we handle only
+                // accounts with no extra signers at present.
+                if (iter.entry().data.account().signers.empty())
+                {
+                    AccountsToUpsert.push_back(iter.entry());
```

Comment:
as keys are large, this may add non trivial overhead to ledgers that have a lot of ledger entries (related: the change https://github.com/stellar/stellar-core/pull/1871 turned a 5s ledger into sub second - all because of the change to keys); this is especially true for offers.


I wonder if we could instead use iterators for doing this as we are not doing anything interesting in accumulate other than very cheap filtering.

Reply:
Yeah, for the initial cut I wanted to be minimally invasive; a slightly-more-invasive version would hold copies of the iterators aside. I'm not sure it'll really be visible against the cost of growing giant string-y array literals to send as query parameters, but I'll do some measuring.

(Looking at #1871 I'm a little nervous about that -- did you ensure nothing map-order-sensitive was perturbed? Some things really like working over sets/maps in deterministic & sorted order.)

Reply:
fixed.